sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -144,7 +153,22 @@
             if (options.getExecutorService() != null) {
                 publicClientApplicationBuilder.executorService(options.getExecutorService());
             }
+            if (options.isSharedTokenCacheEnabled()) {
+                try {
+                    publicClientApplicationBuilder.setTokenCacheAccessAspect(
+                            new PersistenceTokenCacheAccessAspect(options.getPersistenceSettings()));
+                } catch (Throwable t) {
+                    if (sharedTokenCacheCredential) {
+                        throw logger.logExceptionAsError(new CredentialUnavailableException(
+                            "Shared token cache is unavailable in this environment.", t));
+                    } else {
+                        throw logger.logExceptionAsError(new ClientAuthenticationException(
+                            "Shared token cache is unavailable in this environment.", null, t));
```

Comment:
Is this exception message meant to be the same as the one above?

Reply:
They are indeed the same - just wrapping in different exceptions to be handled differently downstream.